File: 168-excel-sheet-column-title/excel-sheet-column-title.py
class Solution:
    def convertToTitle(self, columnNumber: int) -> str:
        if columnNumber <= 0:
            return ""
        SIZE = 26
        number = columnNumber
        result_chars = []
        while number > 0:
            remainder = (number - 1) % SIZE
            quotient = (number - 1) // SIZE
            print(quotient)
            result_chars.insert(0, chr(65 + remainder))
            number =  quotient
        
        return "".join(result_chars)
            

File: 168-excel-sheet-column-title/excel-sheet-column-title.java
class Solution {
    public String convertToTitle(int columnNumber) {
        if(columnNumber <= 0){
            return "";
        }
        final int SIZE = 26;
        StringBuilder resultBuilder = new StringBuilder();

        while(columnNumber > 0){
            int remainder = (columnNumber - 1) % SIZE;
            int quotient = (columnNumber - 1) / SIZE; 
            resultBuilder.append((char) ('A' + remainder));
            columnNumber = quotient;
        }
        
        return resultBuilder.reverse().toString();
    }
}